### ISSUE TITLE
Try Storybook

### ISSUE THREAD

**Issue by kueda** [inaturalist/iNaturalistReactNative#153]
Storybook is a very different approach to UI development than what we're doing now, but I think it's worth at least a day of exploration. Try to integrate storybook into our code and use it to redo something relatively simple, like the login screen. Prepare to make a recommendation as to whether it's worth continuing to work with.

**Comment by albullington** [comment]
After a partial day of exploration, I’m not convinced this is a great UI solution for us. Some reasons:

1. It doesn’t seem to be well documented. I tried the automatic setup & couldn’t get that running, so I switched to setting things up manually. Example docs and repos tend to have a lot of outdated ways of doing things in RN like PropTypes and also come with a lot of eslint errors that we need to disable.
2. Storybook renders in the place where I would expect the iNat Mobile app to render. It looks like some devs have gotten around this issue by conditionally rendering either Storybook or their own app. I think this would change our workflow pretty significantly.
3. Stories feel a bit to me like the Redux way of doing things, where you create a component and then assign it different actions. That means if we wanted a button to change color when it gets pressed, for example, stories would work well. But if we want a button to simply show up as disabled when disabled props are passed to it, I’m not sure there’s any advantage to doing that with Storybook.
4. My dev environment slowed down to the point where I couldn’t add an eslint-disabled line to index.js and save the file — saving the file seems to be pretty impossible, so I ended up skipping pre-commit hooks to commit code. I don’t know if this is directly related to Storybook, but it wasn’t happening before.

It is kind of nice to have a preview of the different ways a button could look (primary, neutral, warning), but I don’t think it’s worth the time investment it would take us to rebuild every component.

Initial exploration of styling buttons is in the `storybook` branch.
